Fixes box-shadow serialization order

niki-04 opened the pull request:
<!-- Please describe your changes on the following line: -->
Box shadow in now serialized properly. The changes are in the file  components/style/properties/longhand/effects.mako.rs. Tests for the serialization of box-shadow are in tests/unit/style/properties/serialization.rs. ./mach test-unit -p style passes all tests including the newly added test case.

---
<!-- Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `__` with appropriate data: -->
- [ ] `./mach build -d` does not report any errors
- [X ] `./mach test-tidy` does not report any errors
- [ X] These changes fix #15203  (github issue number if applicable).

<!-- Either: -->
- [X ] There are tests for these changes OR
- [ ] These changes do not require tests because _____

<!-- Pull requests that do not address these steps are welcome, but they will require additional verification as part of the review process. -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/15664)
<!-- Reviewable:end -->

framlog committed:
fixed the wrong behavior of border-spacing

gregkatz committed:
Eliminate a mem::transmute in CodeGen

gw3583 committed:
Update WR (clip scroll group work).

This shouldn't have any noticeable change on Servo, but as we land
various parts of the clip/scroll changes, keeping Servo up to date
will hopefully help catch any regressions early.

niki-04 committed:
value of box-shadow is now serialized properly

niki-04 committed:
value of box-shadow is now serialized properly

niki-04 committed:
value of box-shadow is now serialized properly

niki-04 committed:
box shadow is serialized properly and also added a test for the same

niki-04 committed:
box shadow is serialized properly and also added a test for the same

highfive commented:
Thanks for the pull request, and welcome! The Servo team is excited to review your changes, and you should hear from @simonsapin (or someone else) soon.

highfive commented:
Heads up! This PR modifies the following files:
 * @bholley: components/style/properties/longhand/effects.mako.rs
 * @emilio: components/style/properties/longhand/effects.mako.rs

jdm commented on tests/unit/style/properties/serialization.rs:
```diff
@@ -266,7 +267,6 @@ mod shorthand_serialization {
             let serialization = shorthand_properties_to_string(properties);
             assert_eq!(serialization, "border-style: solid dotted;");
         }
-    }
```
This change appears to be unintentional (and likely breaks the build).

niki-04 replied:
I fixed that error but still the build fails

niki-04 committed:
fixed a bracket error

niki-04 committed:
fixed a bracket error

jdm commented on tests/unit/style/properties/serialization.rs:
```diff
@@ -1114,4 +1114,31 @@ mod shorthand_serialization {
+    }
+
+
+    }
```
This extra closing brace looks like a likely cause of the failing build?

niki-04 replied:
Is there a way to get to know what is causing the build to fail. I removed that bracket it still fails. I don't want to keep bugging you so was wondering if I can fix the build errors.

nox replied:
Building locally will give you any compilation error.

jdm replied:
Additionally, it's hard to provide useful feedback right now because these changes are based on an older version of master that conflicts, so our tools can't even try to build them and report errors right now. You will need to [rebase](https://github.com/servo/servo/wiki/Beginner's-guide-to-rebasing-and-squashing#rebasing) your changes on top of a more recent revision to address this.

niki-04 committed:
removed an extra curly bracket

niki-04 committed:
removed an extra curly bracket

Commit:
Auto merge of #15654 - glennw:update-wr-clip-scroll, r=metajack

Update WR (clip scroll group work).

This shouldn't have any noticeable change on Servo, but as we land
various parts of the clip/scroll changes, keeping Servo up to date
will hopefully help catch any regressions early.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/15654)
<!-- Reviewable:end -->

gw3583 committed:
Update WR (more scroll/clip work, fix for zero sized stacking contexts).

samliu committed:
Replace uses of spidermonkey-specific JS_GetArrayBufferViewType with ArrayBufferView impl's method get_array_type()

Commit:
Auto merge of #15667 - glennw:update-wr-empty-layer-fix, r=shinglyu

Update WR (more scroll/clip work, fix for zero sized stacking contexts).

<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/15667)
<!-- Reviewable:end -->

Commit:
Auto merge of #15612 - UnICorN21:master, r=canaltinova

fixed the wrong behavior of border-spacing

fixed the wrong behavior of border-spacing
---
<!-- Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `__` with appropriate data: -->
- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #15489.

<!-- Either: -->
- [X] There are tests for these changes OR
- [ ] These changes do not require tests because _____

<!-- Pull requests that do not address these steps are welcome, but they will require additional verification as part of the review process. -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/15612)
<!-- Reviewable:end -->

nox committed:
Remove unused features from webvr traits

nox committed:
Remove a useless cfg attribute

nox committed:
Kill the plugins crate and its clippy support

Sometimes clippy gets outdated by months, and its current support setup
means that each Servo component need to opt into it by depending on
the plugins crate manually, and not all components do that.

Commit:
Auto merge of #15659 - gregkatz:eliminate_transmute_find_enum, r=Ms2ger

Eliminate a mem::transmute in CodeGen

<!-- Please describe your changes on the following line: -->
Eliminate a mem::transmute in CodeGen by changing the find_enum_string_index function to take a slice of pairs and return an enum value.

---
<!-- Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `__` with appropriate data: -->
- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #15587 (github issue number if applicable).

<!-- Either: -->
- [ ] There are tests for these changes OR
- [X] These changes do not require tests because _____

<!-- Pull requests that do not address these steps are welcome, but they will require additional verification as part of the review process. -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/15659)
<!-- Reviewable:end -->

Commit:
Auto merge of #15670 - nox:die-plugins-die, r=Manishearth

Kill the plugins crate and its clippy support

<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/15670)
<!-- Reviewable:end -->

Commit:
Auto merge of #15653 - samliu:cleanup_arraybufferview_type, r=cbrewster

Cleanup arraybufferview type

<!-- Please describe your changes on the following line: -->
Replace uses of spidermonkey-specific JS_GetArrayBufferViewType with ArrayBufferView impl's method get_array_type().

Tests pass:
./mach test-wpt tests/wpt/web-platform-tests/WebCrypto
./mach test-wpt tests/wpt/web-platform-tests/webgl

---
<!-- Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `__` with appropriate data: -->
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #15605 (github issue number if applicable).
- [x] There are tests for these changes

<!-- Pull requests that do not address these steps are welcome, but they will require additional verification as part of the review process. -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/15653)
<!-- Reviewable:end -->

niki-04 committed:
rebasing

niki-04 committed:
rebasing

KiChjang commented:
Looks like rebasing went wrong here...

niki-04 commented:
What exactly went wrong in rebasing. I see the checks are still pending.

KiChjang commented:
You shouldn't have other people's commits on your branch.

niki-04 commented:
So what should I do to fix it?

KiChjang commented:
I think you should create a new branch that is updated to the latest master, cherry-picking the commits that are relevant to the issue you're trying to address, and recreate a pull request based off on that branch. Does that make sense?

In general, it is good practice to work on feature branches instead of the master branch, since your master is constantly going through updates from upstream and is very easy to shoot yourself in the foot during rebasing.

SimonSapin commented:
@niki-04 I’ve changed you pull request’s title to explain what it does (or what I assume it does based on the title of the issue it’s fixing, I haven’t looked at the code changes yet). Please do the same if future pull requests you open, thanks! "Fixes #15203" by itself without context is not very informative in the middle of other email notifications :)

niki-04 commented:
@SimonSapin Sorry about that. I would make sure for future pull requests titles are more informative. Also since rebasing went wrong here as suggested by @KiChjang should I open a new pull request.

jdm commented:
A new pull request would not hurt, and it will be easier than trying to force push to this one from a different branch.

niki-04 commented:
I created a new pull request #15703 for the same issue. Hopefully, that's better

jdm commented:
Closing in favour of that one, in that case.
